querydsl-core/src/main/java/com/querydsl/core/types/dsl/PathBuilder.java

```diff
@@ -125,11 +128,12 @@ public Object getMetadata(Path<?> property) {
      */
     @SuppressWarnings("unchecked")
     public PathBuilder<Object> get(String property) {
-        PathBuilder<Object> path = (PathBuilder) properties.get(property);
+        SimpleEntry<String, Class<?>> entry = new SimpleEntry<String, Class<?>>(property, Object.class);
+        PathBuilder<Object> path = (PathBuilder<Object>) properties.get(entry);
         if (path == null) {
             Class<?> vtype = validate(property, Object.class);
             path = new PathBuilder<Object>(vtype, forProperty(property), validator);
-            properties.put(property, path);
+            properties.putIfAbsent(entry, path);
```

Comment:
In [`ConcurrentMap` there is `putIfAbsent`](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentMap.html#putIfAbsent(K,%20V)).
Otherwise animal-sniffer would have complained.